src/api/auth/strategy/jwt.strategy.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { JwtName, JwtPayloadT } from '../types/types';
+import { UserRepository } from 'src/api/user/user.repository';
+import { EnvVariable } from 'src/shared/types/env';
+import { UserDocumentWithoutPassword } from 'schemas/user.schema';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, JwtName.JWT) {
+  constructor(
+    config: ConfigService,
+    private userRepository: UserRepository,
+  ) {
+    super({
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+      secretOrKey: config.get(EnvVariable.JWT_SECRET),
+    });
+  }
+
+  validate(payload: JwtPayloadT): Promise<UserDocumentWithoutPassword> {
+    return this.userRepository.getById(payload.sub);
```

Comment:
Are you sure you want to query db on each request?

Reply:
I would avoid storing email in jwt claims as its violation security policies in EU/USA. Just store user id and if its not found during your flow in controllers/service - just throw an error